### PR TITLE
Use ampdoc API to check the tree position of the amp-animation element

### LINKED ISSUE #11522: amp-animation broken, shows warning when running with amp-shadow
[From Slack](https://amphtml.slack.com/archives/C0ADR0A0K/p1506945593000233)

`amp-animation` checks if the `<amp-animation>` element is a child of the document body. It doesn't use the `ampdoc` APIs for the check, so when it runs in a shadow context the check returns incorrect results:
https://github.com/ampproject/amphtml/blob/master/extensions/amp-animation/0.1/amp-animation.js#L79

The assert check makes amp-animation not work in shadow document.

### PR DESCRIPTION
**Pull request by cvializ**
Fixes #11522 

/to @aghassemi @dvoytenko

### CODE REVIEW

**Comment by cvializ** [comment]
@dvoytenko The test is failing because for FIE the body according to the ampdoc is the body outside the iframe. Do you have any insight on how to handle the FIE case?

File: extensions/amp-animation/0.1/amp-animation.js [hunk 1]
```diff
@@ -73,11 +73,13 @@ export class AmpAnimation extends AMP.BaseElement {
...
   /** @override */
   buildCallback() {
+    const ampdoc = this.getAmpDoc();
+
     user().assert(isExperimentOn(this.win, TAG),
         `Experiment "${TAG}" is disabled.`);
 
     // TODO(dvoytenko): Remove once we support direct parent visibility.
-    user().assert(this.element.parentNode == this.element.ownerDocument.body,
+    user().assert(this.element.parentNode == ampdoc.getBody(),
```
Comment (dvoytenko) [inline extensions/amp-animation/0.1/amp-animation.js]: We'll have to do both for now. E.g. 
```
user().assert(
  this.element.parentNode == this.element.ownerDocument.body ||
  this.element.parentNode == ampdoc.getBody(),
  ....)
```

Comment (cvializ) [reply extensions/amp-animation/0.1/amp-animation.js]: Ah that figures. Thanks!

File: extensions/amp-animation/0.1/amp-animation.js [hunk 2]
```diff
@@ -73,11 +73,15 @@ export class AmpAnimation extends AMP.BaseElement {
...
+    const ampdoc = this.getAmpDoc();
+
     user().assert(isExperimentOn(this.win, TAG),
         `Experiment "${TAG}" is disabled.`);
 
     // TODO(dvoytenko): Remove once we support direct parent visibility.
-    user().assert(this.element.parentNode == this.element.ownerDocument.body,
+    user().assert(
+        this.element.parentNode == this.element.ownerDocument.body ||
+        this.element.parentNode == ampdoc.getBody(),
```
Comment (jridgewell) [inline extensions/amp-animation/0.1/amp-animation.js]: Can't the previous be removed?

Comment (dvoytenko) [reply extensions/amp-animation/0.1/amp-animation.js]: Not yet. The first one covers FIE and the second one covers PWA shadow doc. The first one will be gone as soon as all FIE are on ampdocs.

**Pull request merged**